### PR TITLE
spike java wrapper using JNA

### PR DESCRIPTION
**Pull request by dacolgit**
java C wrapper for windows  based on JNA library
The native code is using iothub nuget packages compiled into a native C dll which exposes the interfaces for the java import lib

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dacolgit__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@dacolgit__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubMessageCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.MQTT_Protocol;
+
+
+public final class DeviceClient 
+{
+	private static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	private IOTHUB_CLIENT_TRANSPORT_PROVIDER transport;
+	
+	protected DeviceClient()
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Why is this needed?

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT 
+	} 
+
+	public DeviceClient(String connString, IotHubClientProtocol protocol)
+	{
+		switch (protocol) 
+		{
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Consider fixing indentation.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT 
+	} 
+
+	public DeviceClient(String connString, IotHubClientProtocol protocol)
+	{
+		switch (protocol) 
+		{ 
+			case HTTPS:
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Please add unit tests. For example, please follow the existing testing from the Java client. Feel free to reach out to Ilie with any questions. The c-code should be mocked and obviously not needed to be tested.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		if (handle == null)
+		{
+			throw new IllegalStateException(
+                    "Cannot close ");
+		}
+	}
+	
+	public void sendEventAsync(Message message,
+	            IotHubEventCallback callback,
+	            Pointer lockobj)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Consider replacing this with Object. Also this is callbackconext not lockobj

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+		    default: 
+		    // should never happen. 
+		        throw new IllegalStateException( 
+		       		"Invalid client protocol specified."); 
+		} 
+
+		handle = Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_CreateFromConnectionString(connString, transport);
+	}
+	
+	public void open()
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Let's please mimic the "c-api". So in that case, we can drop the Open & Close.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+		if (handle == null)
+		{
+			throw new IllegalStateException(
+                    "Cannot send event from "
+                            + "an IoT Hub client that is closed.");
+		}
+			
+		status = Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_SendEventAsync(handle.getPointer(), message.getMessageHandle(), callback, lockobj);
+		
+		if (status != IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_OK)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Modify the API to return the error codes and return status directly.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
+// Copyright(c) Microsoft.All rights reserved.
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Let's follow the c-api instead for this. This will simplify the binding, get rid of converting error codes to exceptions and  remove un-implemented functions (like Open() & Close())

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+		    case MQTT: 
+		    	this.transport = new MQTT_Protocol();
+		        break; 
+		    default: 
+		    // should never happen. 
+		        throw new IllegalStateException( 
+		       		"Invalid client protocol specified."); 
+		} 
+
+		handle = Iothub_client_wrapperLibrary.INSTANCE.IoTHubClient_CreateFromConnectionString(connString, transport);
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: Missing check of handle.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    {
+		int status = IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_ERROR;
+		
+		if(optionName == null)
+        {
+            throw new IllegalArgumentException("optionName is null");
+        }
+        else
+        {
+        	if (this.transport.getClass() == HTTP_Protocol.class)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: c will propagate this automatically to the proper protocol. No need to do this here.

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+ * An IoT Hub status code. Included in a message from an IoT Hub to a device.
+ */
+public enum IotHubStatusCode
+{
+	OK, OK_EMPTY, BAD_FORMAT, UNAUTHORIZED, TOO_MANY_DEVICES,
+    HUB_OR_DEVICE_ID_NOT_FOUND,
+    PRECONDITION_FAILED, REQUEST_ENTITY_TOO_LARGE, THROTTLED,
+    INTERNAL_SERVER_ERROR, SERVER_BUSY, ERROR;
+
+	public static IotHubStatusCode getStatusCode(int index)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: Please use the built-in method to get the enum.

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+     * @param httpsStatus the HTTPS status code.
+     *
+     * @return the corresponding IoT Hub status code.
+     *
+     * @throws IllegalArgumentException if the HTTPS status code does not map to
+     * an IoT Hub status code.
+     */
+    public static IotHubStatusCode getIotHubStatusCode(int httpsStatus)
+    {
+        // Codes_SRS_IOTHUBSTATUSCODE_11_001: [The function shall convert the given HTTPS status code to the corresponding IoT Hub status code.]
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: This is a copy paste. Remove comment.

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+                iotHubStatus = THROTTLED;
+                break;
+            case 500:
+                iotHubStatus = INTERNAL_SERVER_ERROR;
+                break;
+            case 503:
+                iotHubStatus = SERVER_BUSY;
+                break;
+            default:
+                // Codes_SRS_IOTHUBSTATUSCODE_11_002: [If the given HTTPS status code does not map to an IoT Hub status code, the function return status code ERROR.]
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: Same here. Please remove comments here and everywhere.

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+     * Returns the IoT Hub status code referenced by the HTTPS status code.
+     *
+     * @param httpsStatus the HTTPS status code.
+     *
+     * @return the corresponding IoT Hub status code.
+     *
+     * @throws IllegalArgumentException if the HTTPS status code does not map to
+     * an IoT Hub status code.
+     */
+    public static IotHubStatusCode getIotHubStatusCode(int httpsStatus)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: No need for this. Map the error codes from c client.

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java [hunk 14]
```diff
@@ -0,0 +1,445 @@
...
+	public static final Iothub_client_wrapperLibrary INSTANCE = (Iothub_client_wrapperLibrary)Native.loadLibrary(Iothub_client_wrapperLibrary.JNA_LIBRARY_NAME, Iothub_client_wrapperLibrary.class);
+	
+	/*
+	 *				iothub_client.h
+	 */
+	
+	/*
+	 * Original signature : IOTHUB_CLIENT_HANDLE IoTHubClient_CreateFromConnectionString(const char*, IOTHUB_CLIENT_TRANSPORT_PROVIDER)
+	 */
+	IOTHUB_CLIENT_HANDLE IoTHubClient_CreateFromConnectionString(String connectionString, Object protocol);
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: for consistency let's add the public keyword.

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java [hunk 15]
```diff
@@ -0,0 +1,445 @@
...
+import com.sun.jna.ptr.ByReference;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.Library;
+
+/*
+ * JNA Wrapper for library iothub_client_python
+ */
+import com.sun.jna.ptr.PointerByReference;
+
+public interface Iothub_client_wrapperLibrary extends Library{
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: Is this generated code?

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java [hunk 16]
```diff
@@ -0,0 +1,445 @@
...
+        public time_t(long value) { super(Native.LONG_SIZE, value); }
+    }
+	
+	public class StringByReference extends ByReference 
+	{
+	    public StringByReference() {
+	        this(0);
+	    }
+
+	    public StringByReference(int size)
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: Missing tests.

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java [hunk 17]
```diff
@@ -0,0 +1,445 @@
...
+	
+	public class StringByReference extends ByReference 
+	{
+	    public StringByReference() {
+	        this(0);
+	    }
+
+	    public StringByReference(int size) 
+	    {
+	        super(size < 4 ? 4 : size);
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: No argument checking.

File: javawrapper/device/iothub-client-javawrapper/Map.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
+// Copyright(c) Microsoft.All rights reserved.
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Map.java]: Why is this class even needed? Are you using this somewhere?

File: javawrapper/device/iothub-client-javawrapper/Message.java [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+	}
+    
+	public IOTHUB_MESSAGE_HANDLE clone()
+	{
+		return Iothub_client_wrapperLibrary.INSTANCE.IoTHubMessage_Clone(messageHandle);
+	}
+    
+	public byte[] getBytes()
+	{
+		String body;
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Message.java]: Variable not needed. Return from your call directly (.getbytes())

File: javawrapper/device/iothub-client-javawrapper/Message.java [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+			throw new IllegalArgumentException("Property name cannot be 'null'.");
+		}
+
+		if (value == null) 
+		{
+			throw new IllegalArgumentException("Property value cannot be 'null'.");
+		}
+
+		propMap = this.getProperties();
+		Map map = new Map(propMap);
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/Message.java]: Every time you're adding a new property you're creating a new Map. Shouldn't this be the same Map?

File: javawrapper/device/samples/direct call of wrapped functions/Main.java [hunk 21]
```diff
@@ -0,0 +1,164 @@
+package javaWrapper;
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: Folder structure appear with no underlines (maybe a GitHub thing) so confirming.

File: javawrapper/device/samples/direct call of wrapped functions/Main.java [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+	
+	public static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	public static IOTHUB_MESSAGE_HANDLE messageHandle;
+	public static int IOTHUB_CLIENT_OK = 0;
+	static final int MESSAGE_COUNT = 5;
+	public static EVENT_INSTANCE[] messages = new EVENT_INSTANCE[MESSAGE_COUNT];
+	static Pointer receiveContext;
+	
+	
+	public static Iothub_client_wrapperLibrary IoTHub = Iothub_client_wrapperLibrary.INSTANCE;
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: Should not have to do that in the sample. Should be able to call the Client constructor which instantiates this underneath.

File: javawrapper/device/samples/direct call of wrapped functions/Main.java [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+	
+	
+	public static void main(String[] args) throws InterruptedException 
+	{
+		String connectionString = "HostName=javaTest.azure-devices.net;DeviceId=Device1;SharedAccessKey=pXuifS+EUVY7oewL1hGpttpq5r/jmgONjn41xPK5knM=";
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: Please remove all credentials.

File: javawrapper/device/samples/direct call of wrapped functions/Main.java [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+			    if ((messages[i].messageHandle = IoTHub.IoTHubMessage_CreateFromString(message)) == null)
+			    {
+			        System.out.println("ERROR: iotHubMessageHandle is NULL!\r\n");
+			    }
+			    else
+			    {
+			    	MAP_HANDLE propMap;
+			    	
+			    	messages[i].messageTrackingId = i;
+			    	messages[i].write();
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: What is this for?

File: javawrapper/device/samples/direct call of wrapped functions/Main.java [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+			    	propMap = IoTHub.IoTHubMessage_Properties(messages[i].messageHandle);
+			    	String propText = String.format("PropMsg_%d", i);
+			    	if (IoTHub.Map_AddOrUpdate(propMap, "PropName", propText) != 0)
+	                {
+			    		System.out.println("ERROR: Map_AddOrUpdate Failed!\r\n");
+	                }
+	  	
+			    	if (IoTHub.IoTHubClient_SendEventAsync(handle.getPointer(), messages[i].messageHandle, SendConfirmationCallback, messages[i].getPointer()) != IOTHUB_CLIENT_RESULT.IOTHUB_CLIENT_OK)
+			    	{
+			    		System.out.println("ERROR: IoTHubClient_LL_SendEventAsync..........FAILED!\r\n");
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: There is no _LL_. Please update the output messages.

File: javawrapper/device/samples/direct call of wrapped functions/readme.md [hunk 26]
```diff
@@ -0,0 +1,7 @@
+
+# Run sample for windows
+
+- open the Visual Studio project iot_hub_client_dll in \javawrapper\windows and compile it.
+- create an eclipse project and add Main.java and all the .java files in \iothub-client-javawrapper
+- add the jna.jar library
+- copy in the java directory project all the .dll files obtained from the iot_hub_client_dll project
```
Comment (tameraw) [inline javawrapper/device/samples/direct call of wrapped functions/readme.md]: Can you please add missing steps? Appears incomplete.

File: javawrapper/device/samples/send-event/readme.md [hunk 27]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (tameraw) [inline javawrapper/device/samples/send-event/readme.md]: Consolidate all the readme.md files and place in root.

File: javawrapper/device/samples/send-receive-sample/SendReceive.java [hunk 28]
```diff
@@ -0,0 +1,286 @@
...
+             {
+                 Message msg = new Message(msgStr);
+                 msg.setProperty("messageCount", Integer.toString(i));
+                 System.out.println(msgStr);
+                 EventCallback eventCallback = new EventCallback();
+                 Pointer ii = new Pointer(i);
+                 client.sendEventAsync(msg, eventCallback, ii);
+             }
+             catch (Exception e)
+             {
```
Comment (tameraw) [inline javawrapper/device/samples/send-receive-sample/SendReceive.java]: Please print out exception.

File: javawrapper/device/samples/send-sample/Main.java [hunk 29]
```diff
@@ -0,0 +1,100 @@
+package javaWrapper;
```
Comment (tameraw) [inline javawrapper/device/samples/send-sample/Main.java]: Let's rename this sample to send-serialized-event. We already have a send-event up there.

File: javawrapper/windows/packages.config [hunk 30]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Apache.QPID.Proton.AzureIot" version="0.9.0.1-preview-003" targetFramework="native" />
```
Comment (tameraw) [inline javawrapper/windows/packages.config]: Please remove dependency on review versions and replace with our latest Nuget packages.

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java [hunk 31]
```diff
@@ -0,0 +1,130 @@
...
+	private IOTHUB_CLIENT_TRANSPORT_PROVIDER transport;
+	
+	protected DeviceClient()
+	{
+	
+	}
+	
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT
```
Comment (tameraw) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: We also have support for AMQP over websockets in C client. We should bring that in as protocol option as well. Including a sample.

File: javawrapper/windows/packages.config (hunk 30 above)
Comment (dacolgit) [inline javawrapper/windows/packages.config]: Done now I'm using latest nuget packages

File: javawrapper/device/samples/send-sample/Main.java (hunk 29 above)
Comment (dacolgit) [inline javawrapper/device/samples/send-sample/Main.java]: Done

File: javawrapper/device/samples/send-receive-sample/SendReceive.java (hunk 28 above)
Comment (dacolgit) [inline javawrapper/device/samples/send-receive-sample/SendReceive.java]: added

File: javawrapper/device/samples/send-event/readme.md (hunk 27 above)
Comment (dacolgit) [inline javawrapper/device/samples/send-event/readme.md]: Moved all the readme.md in root.

File: javawrapper/device/samples/direct call of wrapped functions/readme.md (hunk 26 above)
Comment (dacolgit) [inline javawrapper/device/samples/direct call of wrapped functions/readme.md]: The procedure is changed. Now it uses Maven

File: javawrapper/device/samples/direct call of wrapped functions/Main.java (hunk 25 above)
Comment (dacolgit) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: Updated the output message

File: javawrapper/device/samples/direct call of wrapped functions/Main.java (hunk 24 above)
Comment (dacolgit) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: I need it to write the fields of the structure to native memory.
You will have to manually call Structure.write/read before/after your native function call, since JNA will not know that the pointer reference a structure that need to be synched with native memory

File: javawrapper/device/samples/direct call of wrapped functions/Main.java (hunk 23 above)
Comment (dacolgit) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: Removed

File: javawrapper/device/samples/direct call of wrapped functions/Main.java (hunk 21 above)
Comment (dacolgit) [inline javawrapper/device/samples/direct call of wrapped functions/Main.java]: I added underlines to folder structure

File: javawrapper/device/iothub-client-javawrapper/Message.java (hunk 19 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Message.java]: Yes, removed variable

File: javawrapper/device/iothub-client-javawrapper/Message.java (hunk 20 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Message.java]: You are right. I need to create a map when I create a new message and not when I set a property.

File: javawrapper/device/iothub-client-javawrapper/Map.java (hunk 18 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Map.java]: I need it in the message class, in the setProperty method

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java (hunk 17 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: added checking

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java (hunk 15 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: No. I just used the generated code as a reference, and I had to modify it by hand

File: javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java (hunk 14 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/Iothub_client_wrapperLibrary.java]: Done

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java (hunk 11 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: Removed

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java (hunk 12 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: removed

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java (hunk 13 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: ok

File: javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java (hunk 10 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/IotHubStatusCode.java]: I'm now using built-in method values() to get the enum

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java (hunk 8 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: added

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java (hunk 6 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: ok

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java (hunk 9 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: removed the if statement

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java (hunk 4 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: replaced with object

File: javawrapper/device/iothub-client-javawrapper/DeviceClient.java (hunk 2 above)
Comment (dacolgit) [inline javawrapper/device/iothub-client-javawrapper/DeviceClient.java]: It seems a git problem, because my code has the right indentation

**Comment by tameraw** [comment]
@dacolgit - Closing as we're handling this via CodeFlow.
